src/main/java/jeeves/parser/Parser.java

```diff
@@ -0,0 +1,221 @@
+            checkIfDescMissing(desc);
+            args.add(desc);
+        } else if (input.startsWith("deadline ")) {
+            args.add("deadline");
+            // Checks if the user has entered the command properly
+            // and extracts the relevant information to parse
+            // then returns args <command, desc, dyDate>
+            try {
+                int byDateIndex = getByDateIndex(input);
+                String desc = input.substring(FINDCOMMAND_DEADLINE_OFFSET, byDateIndex - 1);
+                String byDate = input.substring(byDateIndex + FINDFIELD_BY_OFFSET);
+                args.add(desc);
+                args.add(byDate);
+            } catch (MissingDescriptionException | MissingByException e) {
+                System.out.println(e.getMessage());
+            }
+        } else if (input.startsWith("event ")) {
+            args.add("event");
+            // Checks if the user has entered the command properly
+            // and extracts the relevant information to parse
+            // then returns args <command, desc, fromDate, toDate>
+            try {
+                int fromDateIndex = getFromDateIndex(input);
+                int toDateIndex = getToDateIndex(input, fromDateIndex);
+                String desc = input.substring(FINDCOMMAND_EVENT_OFFSET, fromDateIndex - 1);
+                String fromDate = input.substring(fromDateIndex + FINDFIELD_FROM_OFFSET, toDateIndex - 1);
+                String toDate = input.substring(toDateIndex + FINDFIELD_TO_OFFSET);
+                args.add(desc);
+                args.add(fromDate);
+                args.add(toDate);
+            } catch (MissingDescriptionException | MissingFromException | MissingToException e) {
+                System.out.println(e.getMessage());
+            }
+        } else if (input.startsWith("find ")) {
+            // Gets the search term the user wishes to find
+            // and returns args <command, findTerm>
+            args.add("find");
+            String findTerm = input.substring(FINDCOMMAND_FIND_OFFSET);
+            args.add(findTerm);
+        } else if (input.equals("bye")) {
+            args.add(input);
+        }
+        return args;
+    }
+
+    /**
+     * Checks if a string is only made up of integer numbers.
+     * Uses regex to determine if a string contains only integers
+     *
+     * @param input The string to be checked
+     * @return Whether the string is fully comprised of integers
+     */
+    static boolean isNotNumber(String input) {
+        return !input.matches("[0-9]+");
+    }
+
+    /**
+     * Checks if the string provided can be parsed into a valid id Integer.
+     * Throws a custom exception with an appropriate error message upon encountering an error.
+     * Currently, checks if the idString is: Present, is a Number, is within the usable range.
+     *
+     * @param idString The string to be checked.
+     */
+    static void checkIfValidId(String idString) {
```

Comment:
is it better `isValidId()`

Reply:
Attempting to avoid confusing method names as boolean methods typically use the is....() format and this method is void, thanks for the review